src/dotnet/ToolManifest/ToolManifestEditor.cs

```diff
@@ -394,7 +395,7 @@ public string ToJson()
 
                     writer.WriteEndObject();
                     writer.WriteEndObject();
-                    writer.Flush(true);
+                    writer.Flush();
```

Comment:
nit: add the following after the call to flush, if you want to be super paranoid (but it isn't necessary).
```C#
Debug.Assert(writer.CurrentDepth == 0);
```

Reply:
Thanks! I try to minimize debug vs release difference. So I will just keep it like that

Reply:
I don't necessarily see why you wouldn't use `Debug.Asserts` to ensure things you expect to be invariant (just to catch possible oversight in source), but that's fine. Ideally this should be caught in your unit tests anyway.

It's already used all over the place in this repo, which is expected. https://github.com/dotnet/cli/search?q=debug.assert&unscoped_q=debug.assert

Reply:
Btw, for context, the reason why  I brought it up is because `Flush()` doesn't guard against un-closed start object/array delimiters.

So you could have done write.StartObject and then forget to call write.EndObject. Adding such a debug.assert just validates correct depth for you and would catch accidental missing calls (especially if you pass the writer along and have deeper call grabs). FWIW.